### PR TITLE
IGDD-1765 Hub code to support Principals from Certs and JWTs.  Allows for optional mTLS.

### PR DESCRIPTION
**Pull request by pcahillai**
Introducing Principal classes to support certificates and JWT tokens.

pom.xml:
Specified specific version range for springdoc to fix problem with their release candidate build.

src/main/java/gov/cdc/izgateway/db/service/AccessControlService.java:
Code for getting the Principal from RequestContext and performing access control check using roles coming from a JWT token.

src/main/java/gov/cdc/izgateway/db/service/StatusCheckerService.java:
changing method to call setPrincipal instead of setCertificate

src/main/java/gov/cdc/izgateway/principal/provider/CertificatePrincipalProviderImpl.java:
The implementation of the Principal provider for Certificates.

src/main/java/gov/cdc/izgateway/principal/provider/JwtJwksPrincipalProvider.java:
The implementation of the Principal provider for JWTs used with JWKS.

src/main/java/gov/cdc/izgateway/principal/provider/JwtPrincipalProviderConfig.java:
Spring configuration to load the configured JwtPrincipal implementation.

src/main/java/gov/cdc/izgateway/principal/provider/JwtSharedSecretPrincipalProvider.java:
The implementation of the Principal provider for JWTs used with a shared secret.

src/main/java/gov/cdc/izgateway/principal/provider/RoleMapper.java:
Class that will support mapping scopes to roles.

src/main/java/gov/cdc/izgateway/principal/service/HubPrincipalService.java:
Implementation of the PrincipalService interface for Hub.

### CODE REVIEW

File: src/main/java/gov/cdc/izgateway/principal/provider/CertificatePrincipalProviderImpl.java
Comment (keithboone) [inline src/main/java/gov/cdc/izgateway/principal/provider/CertificatePrincipalProviderImpl.java:1]: I think the izgw-hub/gov.cdc.izgateway.principal package belongs in izgw-core/gov.cdc.izgateway.security.principal

File: src/main/java/gov/cdc/izgateway/principal/provider/JwtJwksPrincipalProvider.java
Comment (keithboone) [inline src/main/java/gov/cdc/izgateway/principal/provider/JwtJwksPrincipalProvider.java:1]: I think the izgw-hub/gov.cdc.izgateway.principal package belongs in izgw-core/gov.cdc.izgateway.security.principal

File: src/main/java/gov/cdc/izgateway/principal/provider/JwtPrincipalProviderConfig.java
Comment (keithboone) [inline src/main/java/gov/cdc/izgateway/principal/provider/JwtPrincipalProviderConfig.java:1]: I think the izgw-hub/gov.cdc.izgateway.principal package belongs in izgw-core/gov.cdc.izgateway.security.principal

File: src/main/java/gov/cdc/izgateway/principal/provider/RoleMapper.java
Comment (keithboone) [inline src/main/java/gov/cdc/izgateway/principal/provider/RoleMapper.java:1]: I think the izgw-hub/gov.cdc.izgateway.principal package belongs in izgw-core/gov.cdc.izgateway.security.principal

**Review by keithboone — CHANGES_REQUESTED** [review]
I think the packaging needs work.  Most of the principal related stuff belongs to izgw-core, rather than izgw-hub.

The code isn't wrong, it's just in the wrong place to be taken advantage of by both hub and transform components.

File: src/main/java/gov/cdc/izgateway/principal/provider/CertificatePrincipalProviderImpl.java
Comment (pcahillai) [reply src/main/java/gov/cdc/izgateway/principal/provider/CertificatePrincipalProviderImpl.java:1]: I made the recommended changes.

File: src/main/java/gov/cdc/izgateway/principal/provider/JwtJwksPrincipalProvider.java
Comment (pcahillai) [reply src/main/java/gov/cdc/izgateway/principal/provider/JwtJwksPrincipalProvider.java:1]: I made the recommended changes.

File: src/main/java/gov/cdc/izgateway/principal/provider/JwtSharedSecretPrincipalProvider.java
[the remark this answers is not in the document]
Comment (pcahillai) [reply src/main/java/gov/cdc/izgateway/principal/provider/JwtSharedSecretPrincipalProvider.java:1]: I made the recommended changes.

File: src/main/java/gov/cdc/izgateway/principal/provider/RoleMapper.java
Comment (pcahillai) [reply src/main/java/gov/cdc/izgateway/principal/provider/RoleMapper.java:1]: I made the recommended changes.

File: src/main/java/gov/cdc/izgateway/principal/service/HubPrincipalService.java
[the remark this answers is not in the document]
Comment (pcahillai) [reply src/main/java/gov/cdc/izgateway/principal/service/HubPrincipalService.java:1]: I made the recommended changes.

**Pull request closed without merging**